### PR TITLE
phase1: allow to run webhooks

### PR DESCRIPTION
**Pull request by aparcar**
This is especially nice to trigger Docker builds containing the latest containers.

### CODE REVIEW

File: phase1/config.ini.example [hunk 1]
```diff
@@ -17,6 +17,10 @@ channel = #example-channel
 nickname = example-builder
 password = example
 
+#[webhook]
```
Comment (ynezz) [inline phase1/config.ini.example:20]: Can we make this future proof and allow more hooks from the start, thus handling it as branches/workers?

**Comment by aparcar** [comment]
@ynezz I tried a more generic approach. Sadly I don't have any test infra for now, maybe @f00b4r0 can have a look at this, too?

**Comment by f00b4r0** [comment]
@aparcar not much time right now, however I have a couple questions:
- wouldn't http://docs.buildbot.net/current/manual/configuration/reporters/github_status.html work here?
- extracting branch/target by splitting the builder name is definitely not robust and will be a problem if later we decide to change the naming pattern. These properties should be accessible directly from the build object. Maybe webhooks even supports Interpolate()?
HTH

File: phase1/config.ini.example (hunk 1 above)
Comment (aparcar) [reply phase1/config.ini.example]: Done with the reimplementation.

**Comment by aparcar** [comment]
> * wouldn't http://docs.buildbot.net/current/manual/configuration/reporters/github_status.html work here?

From my understanding this is for the "checks" under a PR and not exactly what we want here. 

Example:
image

> * extracting branch/target by splitting the builder name is definitely not robust and will be a problem if later we decide to change the naming pattern. These properties should be accessible directly from the build object. Maybe webhooks even supports Interpolate()?

Yes it's a bit awkward, do you mind pointing me to the properties I'm looking for?

**Comment by f00b4r0** [comment]
> > * extracting branch/target by splitting the builder name is definitely not robust and will be a problem if later we decide to change the naming pattern. These properties should be accessible directly from the build object. Maybe webhooks even supports Interpolate()?
> 
> Yes it's a bit awkward, do you mind pointing me to the properties I'm looking for?

The `branch` property is automatically set by buildbot and available from the builder and build properties. Check e.g. https://buildbot.staging.openwrt.org/images/api/v2/builds/5989/properties or look on the web UI at any build for the "Build Properties" tab.

For target, I would simply add a builder property at the BuilderConfig() invocation here:
https://github.com/openwrt/buildbot/blob/280a07b9e6ba928ce5495dc19ed68d17e990396c/phase1/master.cfg#LL1278C3-L1278C164

e.g.:
```
c['builders'].append(BuilderConfig(name=bldrname, workernames=workerNames, factory=factory, tags=[brname,], properties={'target':target}, nextBuild=GetNextBuild, canStartBuild=canStartBuild))
```

You can then Interpolate() both properties in your Reporter.

HTH

**Comment by aparcar** [comment]
@f00b4r0 thanks, I reworked this PR based on your suggestions.

**Comment by aparcar** [comment]
In the meantime solved via this:

https://github.com/openwrt/docker/issues/119#issuecomment-1741517213

**Comment by ynezz** [comment]
> In the meantime solved via this:

Do I understand it correctly, that you don't want to proceed with this? Perhaps close it then? If you still want that included, then please make CI happy and thus we can merge/test/deploy this.

**Comment by aparcar** [comment]
I'm using rsync-xfer instead

**Pull request closed without merging**